spec/controllers/attachments_controller_spec.rb

```diff
@@ -96,34 +109,151 @@
+        }
+
+        expect(document.attachments.count).to eq(2)
+        expect(response).to redirect_to(edit_document_path(
+                                          document_type_slug: document_type_slug,
+                                          content_id_and_locale: "#{document_content_id}:#{document_locale}",
+                                        ))
+      end
+
+      context "update_attachment fails" do
+        it "shows an error message and redirects to edit document attachment page" do
+          allow_any_instance_of(AttachmentsController).to receive(:fetch_document).and_return(document)
+          allow_any_instance_of(Document).to receive(:update_attachment).and_return(false)
+          error_message = "There was an error updating the attachment, please try again later."
+
+          post :update, params: {
+            document_type_slug: document_type_slug,
+            document_content_id_and_locale: "#{document_content_id}:#{document_locale}",
+            attachment_content_id: attachment_content_id,
+            attachment: updated_attachment,
+          }
+
+          expect(flash[:danger]).to eq(error_message)
+          expect(response).to redirect_to(
+            edit_document_attachment_path(
+              document_type_slug: document_type_slug,
+              document_content_id_and_locale: "#{document_content_id}:#{document_locale}",
+              attachment_content_id: attachment_content_id,
+            ),
+          )
+        end
+      end
+    end
+
+    context "only the attachment title is updated" do
+      let(:document) { CmaCase.find(document_content_id, document_locale) }
+      let(:updated_file) { Rack::Test::UploadedFile.new("spec/support/images/updated_cma_case_image.jpg", "mime/type") }
+      let(:updated_attachment) do
+        {
+          title: "updated test attachment upload",
+        }
+      end
+
+      it "updates the attachment title" do
+        allow_any_instance_of(AttachmentsController).to receive(:fetch_document).and_return(document)
+        success_message = "Attachment successfully updated"
+
+        post :update, params: {
+          document_type_slug: document_type_slug,
+          document_content_id_and_locale: "#{document_content_id}:#{document_locale}",
+          attachment_content_id: attachment_content_id,
+          attachment: updated_attachment,
+        }
+
+        expect(flash[:success]).to eq(success_message)
+        expect(response).to redirect_to(edit_document_path(
+                                          document_type_slug: document_type_slug,
+                                          content_id_and_locale: "#{document_content_id}:#{document_locale}",
+                                        ))
+      end
+
+      context "error updating the attachment" do
+        it "shows an error message and redirects to edit attachment page" do
+          allow_any_instance_of(Document).to receive(:save).and_return(false)
```

Comment:
Generally it's better to use `#allow` than `#allow_any_instance_of` where possible as it allows you to mock stub with a lot more specificity 

We can do 

```
 allow(document).to receive(:save).and_return(false)
``` 

here